sig-apps/Charter.md

```diff
@@ -0,0 +1,141 @@
+# SIG Apps Charter
+
+## Scope
+
+SIG Apps Covers deploying and operating applications in Kubernetes with a focus on the application developer and application operator experience.
+
+For example, this includes:
+
+* Discussions on all levels from application development, application testing environments, and applications running in production
+* Discussion of how to define and run apps in Kubernetes
+* Demos of relevant tools and projects
+* Discussion on areas of friction that can lead to suggesting improvements or feature requests
+* Development of Kubernetes primitives (e.g., Workloads API) to enable the application developers and operators. Note, most new types should be created as CRDs during their development phase and may always be ecosystem projects
```

Comment:
Can CRDs be owned by SIG Apps? By definition they are not part of core. 

I could see the argument that they can be useful in experimenting with potential core API changes, but I'm not sure how we guarantee that those experiments are short-lived and do not carry on as opinionated tools. Imagine a scenario where SIG Apps maintained a set of CRDs that were highly opinionated and tied to a specific ecosystem tool. These APIs would never have to go through a full community and API review, but would be "official" recommendations of Kubernetes.

Should those experiments instead happen with core alpha APIs that go through full API reviews?

Reply:
> Should those experiments instead happen with core alpha APIs that go through full API reviews?

This question went before SIG Architecture for a case similar to what you're describing. It was for an object to describe applications and started as a KEP for a new core API. The idea, and even a CRD, came out of the App Def WG. WGs don't own code so it fell to SIG Apps to own this development.

The process used to be for a core API to start out, go through alphas and betas, and then get to GA. That process is changing as part of the goal to have k8s be slower moving "boring" infrastructure. The direction from SIG Architecture was for this to start outside the core, prove itself, and if it makes sense to later revisit it going into Kubernetes.

The Application object hasn't been the only example of this direction over the past several months.

The App Def WG wanted this in the name of application and tool compatibility. For example, one could deploy two apps. One with Helm and one with ksonnet. Those apps could be viewed with Kubernetes Dashboard and Tectonic's console.

When a SIG wants to rally around projects like this there is a [process for them to create and maintain repos](https://github.com/kubernetes/community/blob/master/kubernetes-repositories.md#sig-repositories).

Because this was in the apps space and had people working on various tools – sometimes from competing companies – SIG Apps chose to work on this problem in the name of tooling compatibility. If it works out well it may end up in the core of Kubernetes long term. Instead of going through the previous alpha/beta phases it's a CRD and controller maintained by the SIG.

This new process of CRD/controllers from the outside going into Kubernetes is new enough we've talked about some of the gaps and unknowns that will need to be overcome.

This is an example of processes on Kubernetes changing a little bit. We've talked about this in SIG Apps to try and communicate it over the past several months.